http_request.h

```diff
@@ -0,0 +1,23 @@
+#ifndef HTTP_REQUEST_H
+#define HTTP_REQUEST_H
+
+#include <string>
+#include <vector>
+#include "http_header.h"
+
+
+namespace http {
+
+// HTTP/1.0 request received from a client
+struct request {
+    std::string         method;  // Indicates what is to be performed
+    std::string         uri;     // Identifies the resource for the request
+    std::string         path;    // URL converted to a path
+    std::vector<header> headers; // Headers included in the request
+    std::string         string;  // The entire, unparsed request as a string
```

Comment:
I feel like this can be renamed rather than just string like std::string even though it is part of the struct request and accessed like request.string to not confuse with STL.